modules/koblerBidAdapter.js

```diff
@@ -0,0 +1,226 @@
+import {BANNER} from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'kobler';
+const BIDDER_ENDPOINT = 'https://bid.essrtb.com/bid/prebid_rtb_call';
+const TIMEOUT_NOTIFICATION_ENDPOINT = 'https://bid.essrtb.com/notify/prebid_timeout';
+const DEFAULT_CURRENCY = 'USD';
+const DEFAULT_TIMEOUT = 1000;
+const TIME_TO_LIVE_IN_SECONDS = 10 * 60;
+
+export const isBidRequestValid = function (bid) {
+  return !(!bid || !bid.bidId || !bid.params || !bid.params.placementId);
+};
+
+export const buildRequests = function (validBidRequests, bidderRequest) {
+  return {
+    method: 'POST',
+    url: BIDDER_ENDPOINT,
+    data: buildOpenRtbBidRequestPayload(validBidRequests, bidderRequest),
+    options: {
+      contentType: 'application/json'
+    }
+  };
+};
+
+export const interpretResponse = function (serverResponse) {
+  const res = serverResponse.body;
+  const bids = []
+  if (res) {
+    res.seatbid.forEach(sb => {
+      sb.bid.forEach(b => {
+        bids.push({
+          requestId: b.impid,
+          cpm: b.price,
+          currency: res.cur,
+          width: b.w,
+          height: b.h,
+          creativeId: b.crid,
+          dealId: b.dealid,
+          netRevenue: true,
+          ttl: TIME_TO_LIVE_IN_SECONDS,
+          ad: b.adm,
+          nurl: b.nurl,
+          meta: {
+            advertiserDomains: b.adomain
+          }
+        })
+      })
+    });
+  }
+  return bids;
+};
+
+export const onBidWon = function (bid) {
+  const cpm = bid.cpm || 0;
+  const adServerPrice = utils.deepAccess(bid, 'adserverTargeting.hb_pb') || 0;
+  if (utils.isStr(bid.nurl) && bid.nurl !== '') {
+    const winNotificationUrl = utils.replaceAuctionPrice(bid.nurl, cpm)
+      .replace(/\${AD_SERVER_PRICE}/g, adServerPrice);
+    utils.triggerPixel(winNotificationUrl);
+  }
+};
+
+export const onTimeout = function (timeoutDataArray) {
+  if (utils.isArray(timeoutDataArray)) {
+    timeoutDataArray.forEach(timeoutData => {
+      const query = utils.parseQueryStringParameters({
+        ad_unit_code: timeoutData.adUnitCode,
+        auction_id: timeoutData.auctionId,
+        bid_id: timeoutData.bidId,
+        timeout: timeoutData.timeout,
+        placement_id: utils.deepAccess(timeoutData, 'params.0.placementId'),
+        page_url: window.location.href,
+      });
+      const timeoutNotificationUrl = `${TIMEOUT_NOTIFICATION_ENDPOINT}?${query}`;
+      utils.triggerPixel(timeoutNotificationUrl);
+    });
+  }
+};
+
+function buildOpenRtbBidRequestPayload(validBidRequests, bidderRequest) {
+  const imps = validBidRequests.map(br => buildOpenRtbImpObject(br));
+  const timeout = bidderRequest.timeout || config.getConfig('bidderTimeout') || DEFAULT_TIMEOUT;
+  const pageUrl = (bidderRequest.refererInfo && bidderRequest.refererInfo.referer)
+    ? bidderRequest.refererInfo.referer
+    : window.location.href;
+
+  const request = {
+    id: bidderRequest.auctionId,
+    at: 1, // TODO: is the auction always first price?
```

Comment:
Since we dont do any optimization based for second price or first price, setting it up to 1 has no harm. I guess we are not using this field in bidder execpt logging.

Reply:
Yes, this is more about having the correct understanding of how Prebid auctions work. So I plan on asking this in the PR to the prebid repo (prebid/Prebid.js)

Reply:
okay. Prebid is in general first price.